lasio/las.py

```diff
@@ -139,25 +137,43 @@ def read(
             provisional_wrapped = "YES"
             provisional_null = None
 
-            section_positions = reader.find_sections_in_file(file_obj)
+            section_positions = reader.find_sections_in_file(str(file_ref))
             logger.debug("Found {} sections".format(len(section_positions)))
             if len(section_positions) == 0:
                 raise KeyError("No ~ sections found. Is this a LAS file?")
 
             data_section_indices = []
+            file_obj, self.encoding = reader.open_file(file_ref, **kwargs)
+
             for i, (k, first_line, last_line, section_title) in enumerate(
                 section_positions
             ):
-                section_type = reader.determine_section_type(section_title)
-                logger.debug(
-                    "Parsing {typ} section at lines {first_line}-{last_line} ({k} bytes) {title}".format(
-                        typ=section_type,
-                        title=section_title,
-                        first_line=first_line + 1,
-                        last_line=last_line + 1,
-                        k=k,
+
+                # Translate byte-string to string
+                section_title_str = section_title.decode(self.encoding)
```

Comment:
This ```decode()``` seems like it should be in ```reader.find_sections_in_file()``` but currently we don't have the encoding available in ```reader.find_sections_in_file()```.  So if we wanted to move the ```decode()```, we would have to pass the encoding to ```reader.find_sections_in_file()```.  Let me know if that is this should be changed.